src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROW_INDEX = 1.5;
+    private static final int START_INDEX_VALUE = 0;
+    private T[] dataArray;
+    private int size;
+
+    public ArrayList() {
+        dataArray = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        dataArray = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == dataArray.length) {
+            dataArray = grow();
+        }
+        dataArray[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("index is invalid");
+        }
+        if (size == dataArray.length) {
+            dataArray = grow();
+        }
+        if (index == size) {
+            dataArray[size++] = value;
+        } else {
+            System.arraycopy(dataArray, index, dataArray, index + 1, size++);
```

Comment:
The `System.arraycopy` is used incorrectly here. The fifth parameter should be the number of elements to be copied, which is `size - index` instead of `size++`. Also, incrementing `size` inside the `arraycopy` call can lead to unexpected behavior. Increment `size` after the elements have been shifted.